hazelcast/src/main/java/com/hazelcast/sql/SqlQuery.java

```diff
@@ -0,0 +1,286 @@
+    }
+
+    /**
+     * Sets the SQL query to be executed.
+     * <p>
+     * The SQL query cannot be null or empty.
+     *
+     * @param sql SQL query.
+     * @return This instance for chaining.
+     * @throws NullPointerException If passed SQL query is null
+     * @throws IllegalArgumentException If passed SQL query is empty
+     */
+    @Nonnull
+    public SqlQuery setSql(@Nonnull String sql) {
+        Preconditions.checkNotNull(sql, "SQL cannot be null");
+
+        if (sql.length() == 0) {
+            throw new IllegalArgumentException("SQL cannot be empty");
+        }
+
+        this.sql = sql;
+
+        return this;
+    }
+
+    /**
+     * Gets the query parameters.
+     *
+     * @return Query parameters.
+     */
+    @Nonnull
+    public List<Object> getParameters() {
+        return parameters != null ? parameters : Collections.emptyList();
+    }
+
+    /**
+     * Sets the query parameters.
+     * <p>
+     * You may define parameter placeholders in the query with the {@code "?"} character. For every placeholder, a parameter
+     * value must be provided.
+     * <p>
+     * When the method is called, the content of the parameters list is copied. Subsequent changes to the original list don't
+     * change the query parameters.
+     *
+     * @see #addParameter(Object)
+     * @see #clearParameters()
+     * @param parameters Query parameters.
+     * @return This instance for chaining.
+     */
+    @Nonnull
+    public SqlQuery setParameters(List<Object> parameters) {
+        if (parameters == null || parameters.isEmpty()) {
+            this.parameters = null;
+        } else {
+            this.parameters = new ArrayList<>(parameters);
+        }
+
+        return this;
+    }
+
+    /**
+     * Adds a single parameter to the end of the parameters list.
+     *
+     * @see #setParameters(List)
+     * @see #clearParameters()
+     * @param parameter Parameter.
+     * @return This instance for chaining.
+     */
+    @Nonnull
+    public SqlQuery addParameter(Object parameter) {
+        if (parameters == null) {
+            parameters = new ArrayList<>(1);
+        }
+
+        parameters.add(parameter);
+
+        return this;
+    }
+
+    /**
+     * Clear query parameters.
+     *
+     * @see #setParameters(List)
+     * @see #addParameter(Object)
+     * @return This instance for chaining.
+     */
+    @Nonnull
+    public SqlQuery clearParameters() {
+        this.parameters = null;
+
+        return this;
+    }
+
+    /**
+     * Gets the query timeout in milliseconds.
+     *
+     * @return Query timeout in milliseconds.
+     */
+    public long getTimeout() {
+        return timeout;
+    }
+
+    /**
+     * Sets the query timeout in milliseconds.
+     * <p>
+     * If the timeout is reached for a running query, it will be cancelled forcefully.
+     * <p>
+     * Zero value means no timeout. {@value #TIMEOUT_NOT_SET} means that the value from {@link SqlConfig#getQueryTimeout()} will
```

Comment:
We should also create a constant for the value of 0.

Reply:
Added a constant.